src/ride/ride.c

```diff
@@ -4540,7 +4540,7 @@ train_ref vehicle_create_train(int rideIndex, int x, int y, int z, int vehicleIn
 {
 	rct_ride *ride = get_ride(rideIndex);
 
-	uint8 trainLayout[42];
+	uint8 trainLayout[RIDE_MAX_CARS_PER_TRAIN];
```

Comment:
Is the reason why this had a size of 42 known? It might be handy to add an assert here in case.
`assert(ride->num_cars_per_train <= RIDE_MAX_CARS_PER_TRAIN);`